diff --git a/rocm-systems/projects/rocprofiler-sdk/source/lib/output/CMakeLists.txt b/rocm-systems/projects/rocprofiler-sdk/source/lib/output/CMakeLists.txt
index 9f904cc..d80a45f 100644
--- a/rocm-systems/projects/rocprofiler-sdk/source/lib/output/CMakeLists.txt
+++ b/rocm-systems/projects/rocprofiler-sdk/source/lib/output/CMakeLists.txt
@@ -58,6 +58,8 @@ add_library(rocprofiler-sdk::rocprofiler-sdk-output-library ALIAS
             rocprofiler-sdk-output-library)
 target_sources(rocprofiler-sdk-output-library PRIVATE ${TOOL_OUTPUT_SOURCES}
                                                       ${TOOL_OUTPUT_HEADERS})
+target_include_directories(rocprofiler-sdk-output-library SYSTEM
+                           PRIVATE ${SQLite3_INCLUDE_DIR})
 # targets linking to this library should also link to the
 # rocprofiler-sdk::rocprofiler-sdk-sqlite3 target if the target is a shared library. Do
 # not link to the sqlite3 library here because the Python bindings target should not
diff --git a/rocm-systems/projects/rocprofiler-sdk/source/lib/python/utilities.cmake b/rocm-systems/projects/rocprofiler-sdk/source/lib/python/utilities.cmake
index b633f6b..98e0d31 100644
--- a/rocm-systems/projects/rocprofiler-sdk/source/lib/python/utilities.cmake
+++ b/rocm-systems/projects/rocprofiler-sdk/source/lib/python/utilities.cmake
@@ -195,7 +195,8 @@ function(rocprofiler_rocpd_python_bindings _VERSION)
         PRIVATE libpyrocpd.cpp libpyrocpd.hpp
                 $<TARGET_OBJECTS:rocprofiler-sdk::rocprofiler-sdk-object-library>)
     target_include_directories(rocprofiler-sdk-rocpd-python-bindings-${_VERSION} SYSTEM
-                               PRIVATE ${Python3_INCLUDE_DIRS})
+                               PRIVATE ${Python3_INCLUDE_DIRS}
+                                       ${SQLite3_INCLUDE_DIR})
 
     # do not link to sqlite3 library here. Python will import the _sqlite3 extension
     # module which links to sqlite3, and we want to avoid mixed-lib symbol collisions by
