diff --git a/rocm-systems/projects/clr/opencl/packaging/CMakeLists.txt b/rocm-systems/projects/clr/opencl/packaging/CMakeLists.txt
index 5ae474980..e7272f89c 100644
--- a/rocm-systems/projects/clr/opencl/packaging/CMakeLists.txt
+++ b/rocm-systems/projects/clr/opencl/packaging/CMakeLists.txt
@@ -104,14 +104,7 @@ if(BUILD_ICD)
   set(CPACK_RPM_BINARY_PACKAGE_REQUIRES "comgr, hsa-rocr, rocm-core")
   set(CPACK_RPM_ASAN_PACKAGE_REQUIRES "comgr-asan, hsa-rocr-asan, rocm-core-asan")
 else()
-  set(OCL_ICD_PACKAGE "ocl-icd")
-  if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-    file(READ "/etc/os-release" OS_RELEASE_CONTENT)
-    string(TOLOWER "${OS_RELEASE_CONTENT}" OS_RELEASE_CONTENT_LOWER)
-    if(OS_RELEASE_CONTENT_LOWER MATCHES "suse" OR OS_RELEASE_CONTENT_LOWER MATCHES "sles")
-      set(OCL_ICD_PACKAGE "ocl-icd-devel")
-    endif()
-  endif()
+  set(OCL_ICD_PACKAGE "ocl-icd" CACHE STRING "RPM package providing the OpenCL ICD loader")
   set(CPACK_RPM_BINARY_PACKAGE_REQUIRES "comgr, hsa-rocr, rocm-core, ${OCL_ICD_PACKAGE}")
   set(CPACK_RPM_ASAN_PACKAGE_REQUIRES "comgr-asan, hsa-rocr-asan, rocm-core-asan")
 endif()
